app/components/profile.jsx

```diff
@@ -19,73 +23,60 @@ export class Profile extends Component {
     this.reset();
   }
 
-  componentWillUpdate (props) {
-    this.reset(props);
+  componentWillUpdate ({ params: { username } }) {
+    if (username !== this.props.params.username) {
+      this.reset();
+    }
   }
 
-  reset (props) {
-    const { checkVersion } = props || this.props;
+  reset () {
+    const { checkVersion, params: { username }, retrieveUser, setCurrentUser, setShowShare } = this.props;
```

Comment:
in this line, we're pulling out all of these variables and functions out of `this.props`, including `username`. like i said before, when we call this in `componentWillReceiveProps` this wont be using all the new stuff, itll still be using the old props. so we ned to conditionally use the new props that are being passed in. thats super easy to do. we just need to replace `this.props` with `props || this.props`. so it'll try to use `props` first but if it's `undefined`, itll default to using `this.props`. easy peasy.

final result:

```js
const { checkVersion, params: { username }, retrieveUser, setCurrentUser, setShowShare } = props || this.props;
```

you should also do this in `tracker.jsx` (line 35)